tgs/src/main/cpp/tgs/RandomForest/RandomForest.cpp

```diff
@@ -81,7 +81,7 @@ void RandomForest::trainBinary(const std::shared_ptr<DataFrame>& data, unsigned
 
         std::vector<std::string> badFactors;
 
-        unsigned int cutOffIdx = topFactors.size() - (unsigned int)((double)topFactors.size() * retrain);
+        unsigned long cutOffIdx = topFactors.size() - (unsigned int)((double)topFactors.size() * retrain);
```

Comment:
Again the variable's data type is correct, the result of the calculation needs to be cast to `unsigned int` instead of being an `unsigned long`.